pkg/controller/core/cohort_controller.go

```diff
@@ -110,5 +150,74 @@ func (r *CohortReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		log.V(2).Error(err, "Error adding or updating cohort in the cache")
 	}
 	r.qManager.AddOrUpdateCohort(ctx, &cohort)
-	return ctrl.Result{}, nil
+
+	err := r.updateCohortStatusIfChanged(ctx, &cohort)
+	return ctrl.Result{}, err
+}
+
+func (r *CohortReconciler) updateCohortStatusIfChanged(ctx context.Context, cohort *kueue.Cohort) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	oldStatus := cohort.Status.DeepCopy()
+
+	stats, err := r.cache.CohortStats(cohort)
+	if err != nil {
+		log.Error(err, "Failed getting cohort usage from cache")
```

Comment:
Would it log the cohort key (name / ns) in question? Probably worth adding this to the logger values.

Reply:
Yes, we have this information. This log from context.

```
2025-03-14T15:04:35.173342+02:00	ERROR	core/cohort_controller.go:164	Failed getting cohort usage from cache	{"controller": "cohort", "controllerGroup": "kueue.x-k8s.io", "controllerKind": "Cohort", "Cohort": {"name":"root"}, "namespace": "", "name": "root", "reconcileID": "e80cf601-8791-4bec-ad4e-0db5abb6ead0"}

```